public/pages/CreateTrigger/components/CompositeTriggerCondition/ExpressionBuilder.js

```diff
@@ -0,0 +1,386 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import {
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiPopover,
+  EuiComboBox,
+  EuiButtonIcon,
+  EuiExpression,
+  EuiToolTip,
+} from '@elastic/eui';
+import * as _ from 'lodash';
+import { FormikFormRow, FormikInputWrapper } from '../../../../components/FormControls';
+import { getMonitors } from '../../../CreateMonitor/components/AssociateMonitors/AssociateMonitors';
+
+export const conditionToExpressions = (condition = '', monitors) => {
+  if (!condition.length) return [];
+
+  const conditionMap = {
+    '&&': 'AND',
+    '||': 'OR',
+    '!': 'NOT',
+    '': '',
+    '&& !': 'AND_NOT',
+    '|| !': 'OR_NOT',
+  };
+  const queryToExpressionRegex = new RegExp(
+    /( && || \|\| || && \!|| \|\| \!)?(monitor\[id=(.*?)\])/,
+    'gm'
+  );
+  const matcher = condition.matchAll(queryToExpressionRegex);
+  let match;
+  let expressions = [];
+  let counter = 0;
+  while ((match = matcher.next().value)) {
+    if (counter && !match[1]) return []; // Didn't find condition after the first match
+
+    const monitorId = match[3]?.trim(); // match [3] is the monitor_id
```

Comment:
Is it guaranteed to always have 4 values in `matcher.next().value`? If not, should we do stuff like `match?.[3]`?

Reply:
If the match doesn't have 4 values, we already have the check for the 4th value - `match[3]?.trim()`

Reply:
Yea but if the index `3` is not there, wont it throw array out of bounds exception since we first do `match[3]` and then do the trim call.

Reply:
No, in JS it would simply return `undefined` and not throw exception